src/spn/algorithms/MEU.py

```diff
@@ -0,0 +1,162 @@
+"""
+Created on March 28, 2019
+@author: Hari Teja Tatavarti
+
+"""
+
+from spn.algorithms.MPE import get_node_funtions
+from spn.algorithms.Inference import  likelihood, max_likelihood, log_likelihood, sum_likelihood, prod_likelihood
+from spn.algorithms.Validity import is_valid
+from spn.structure.Base import get_nodes_by_type, Max, Leaf, Sum, Product, get_topological_order_layers
+
+import numpy as np
+from collections import defaultdict
+import collections
+
+
+def merge_input_vals(l):
```

Comment:
What is `l` supposed to be over here, please refrain from using single character variable names.

Reply:
Thanks for pointing it out. It is supposed to be parent_result. I have used the method from MPE.py module as is. You may want to review it there as well.

Reply:
Ah okay, maybe we can change the variable name to be something like `parent_result` or something meaningful?

Reply:
It concatenates `parent_result`, so I think may be using that name makes sense.